hastexo/jobs.py

```diff
@@ -151,6 +153,8 @@ def run(self):
                 suspend_timestamp__lt=cutoff
             ).exclude(
                 status__in=dont_delete
+            ).exclude(
+                provider__exact=''
```

Comment:
Under what circumstances would this happen? In other words, when is the provider ever empty?

Reply:
Great question: I originally thought this would never happen.  However, on versions prior to 2.5.3, unhandled exceptions (such as HTTPExceptions from Heat), could result in a bogus state in the database, including (but not limited to) a supposedly deletable stack without a provider.  I did my best to not let this happen in 2.5.3, but on the off chance I didn't fix all the holes, it's best to avoid the problem altogether "from the other side".

Reply:
Makes sense, thanks!